e2e/tests-dfx/wallet.bash

```diff
@@ -135,39 +133,41 @@ teardown() {
     dfx_new
     dfx_start
     WALLET_ID=$(dfx identity get-wallet)
-    CREATE_RES=$(dfx canister --no-wallet call "${WALLET_ID}" wallet_create_wallet "(record { cycles = (2000000000000:nat64); settings = record {controller = opt principal \"$(dfx identity get-principal)\";};})")
+    CREATE_RES=$(dfx canister call "${WALLET_ID}" wallet_create_wallet "(record { cycles = (2000000000000:nat64); settings = record {controller = opt principal \"$(dfx identity get-principal)\";};})")
     CHILD_ID=$(echo "${CREATE_RES}" | tr '\n' ' ' |  cut -d'"' -f 2)
-    assert_command dfx canister --no-wallet call "${CHILD_ID}" wallet_balance '()'
+    assert_command dfx canister call "${CHILD_ID}" wallet_balance '()'
 }
 
-@test "bypass wallet call as user" {
+@test "forward user call through wallet" {
     dfx_new
     install_asset identity
     dfx_start
-    assert_command dfx canister --no-wallet create --all
+    WALLET=$(dfx identity get-wallet)
+    assert_command dfx canister --wallet "$WALLET" create --all
     assert_command dfx build
-    assert_command dfx canister --no-wallet install --all
+    assert_command dfx canister --wallet "$WALLET" install --all
 
-    CALL_RES=$(dfx canister --no-wallet call e2e_project fromCall)
+    CALL_RES=$(dfx canister --wallet "$WALLET" call e2e_project fromCall)
     CALLER=$(echo "${CALL_RES}" | cut -d'"' -f 2)
-    ID=$(dfx identity get-principal)
-    assert_eq "$CALLER" "$ID"
+    assert_eq "$CALLER" "$WALLET"
 
-    assert_command dfx canister --no-wallet call e2e_project amInitializer
-    assert_eq '(true)'
```

Comment:
Shouldn't we still have a test for this functionality? (create and deploy a canister with no wallet; call it directly)

Reply:
Should we? The only thing that still retains wallet passthrough by default is `create`, which `amInitializer` wouldn't measure; AFAICT the old tests are now indistinguishable from the ones in identity.bash.

Reply:
If we aren't removing test coverage, then it's fine.